services/web_research_service.py

```
@@ -0,0 +1,43 @@
+            results.append({"url": url, "snippet": snippet, "verified": verified})
+        return results
+
+    def fetch_and_extract(self, url: str) -> str:
+        try:
+            resp = requests.get(url, timeout=10)
+            soup = BeautifulSoup(resp.text, "html.parser")
+            # Extract main text (very basic)
+            paragraphs = soup.find_all("p")
+            text = "\n".join([p.get_text() for p in paragraphs])
+            return text[:1000]  # Limit for demo
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)